components/Carousel/Structure.js

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState, useCallback } from "react";
+import { Flex } from "@chakra-ui/react";
+import Content from "./Content";
+
+function Structure({ slides }) {
+  const [currentSlide, setCurrentSlide] = useState(0);
+
+  const slidesCount = slides.length;
+  const prevSlide = useCallback(() => {
+    setCurrentSlide((s) => (s === 0 ? slidesCount - 1 : s - 1));
+  }, [slidesCount]);
+  const nextSlide = useCallback(() => {
+    setCurrentSlide((s) => (s === slidesCount - 1 ? 0 : s + 1));
+  }, [slidesCount]);
+
+  const carouselStyle = {
+    transition: "all .5s",
+    mx: `-${currentSlide * 100}%`,
+  };
+
+  const SLIDES_INTERVAL_TIME = 5000;
```

Comment:
Make the interval and direction as props for the component with default values. it will make the component more dynamic